applications/NeuralNetworkApplication/python_scripts/data_loading_utilities.py

```diff
@@ -0,0 +1,132 @@
+            category
+            + " data format not supported. Supported formats are .dat, .npy, .pkl and .h5"
+        )
+    return data
+
+
+def ImportH5(external_file, lookback):
+    "Import the data of an h5 file generated for neural network training. Check the expected format."
+    print("Warning: HDF5 functionalities are not fully implemented.")
+    if lookback:
+        raw = ListDataWithLookback()
+    else:
+        raw = ListNeuralNetworkData()
+    with h5py.File(external_file, "r") as f:
+        for time in f.keys():
+            time_array = []
+            for model_part in f[time].keys():
+                for variables in f[time][model_part]["NodalSolutionStepData"].keys():
+                    value = f[time][model_part]["NodalSolutionStepData"][variables][:]
+                    time_array.append(value[:])
+            raw.AddToList(np.squeeze(time_array))
+    return raw
+
+
+def ImportAscii(external_file, lookback):
+    "Import the data of an ascii file generated for neural network training."
+    raw = []
+    with open(external_file, "r") as file:
+        for line in file:
+            line = line.strip()
+            if not line:
+                continue
+            raw.append(list(map(str, line.split())))
+    if lookback:
+        raw_splits = ListDataWithLookback()
+    else:
+        raw_splits = ListNeuralNetworkData()
+    for line in raw:
+        raw_splits_line = []
+        for data in line:
+            if not isfloat(data.split(" ")[0]):
+                array = "".join(data[4:-1])
+                array_floats = list(
+                    map(
+                        float,
+                        array.split(
+                            ",",
+                        ),
+                    )
+                )
+                raw_splits_line.extend(array_floats)
+            else:
+                array_floats = float(data)
+                raw_splits_line.append(array_floats)
+        raw_splits.AddToList(np.squeeze(raw_splits_line))
+    return raw_splits
+
+
+def ImportNpy(external_file, lookback):
+    "Import the data from a npy file."
+    if lookback:
+        raw = ListDataWithLookback()
+    else:
+        raw = ListNeuralNetworkData()
+    raw.ImportFromArray(np.squeeze(np.load(external_file)))
+    return raw
+
+
+def ImportPkl(external_file):
+    "Import the data from a pkl file that contains a ListNeurakNetworkData or ListDataWithLookback object."
+    with open(external_file, "rb") as file:
+        raw = pickle.load(file)
+    return raw
+
+
+def ImportDictionaryFromText(external_file):
+    "Import the data from a text file to a dictionary"
+
+    with open(external_file, "r") as file:
+        data = file.read()
+
+    return ast.literal_eval(data)
+
+
+def UpdateDictionaryJson(external_file, dump_data):
+    with open(external_file, "w") as f:
+        f.seek(0)
+        json.dump(dump_data, f)
+        f.truncate()
+
+
+def KratosVectorToList(vector):
```

Comment:
I think it will be interestinmg to implement this in C++, the implementation would be trivial

Reply:
well, i think that 
~~~py

my_list = list(input_kratos_vector) )
~~~

should actually work. 

other than that you can interprete it as a numpy array (with copy=False) to avoid copies, and then you can do with it anything you can do with a numpy array